### PR TITLE
Adding neighbor Finished

### PR DESCRIPTION
**Pull request by wfystx**
- Exposed port 8001 for cpisync. 8002 for tcp
- Simplified GossipNode's construtor.
  - Added a NetworkUtils.h which contains a method to get the hostname and IP address of current node. With example code of how to use it. See README in src/common/utils
  - DB path is also automatically set.
- Added several functions in GossipNode.h used for adding neighbors.
- Modifed niok.cc, improved user interface. **More importantly,** now is using two threads, one for main funcion(cli), the other one for listenning TCP connection.
- Also added unit test for NetworkUtils. And enhanced TCPServer.h and TCPClient.h

### CODE REVIEW

**Comment by aktrinh17** [comment]
Looks good. Great work!

File: src/CPISync/GossipNode.cc [hunk 1]
```diff
@@ -165,18 +184,59 @@ void GossipNode::processLogEntry()
...
     EOC = logSize;
 }
 
-void GossipNode::addNeighbor(IPv4 *ip)
+void GossipNode::listenTCP()
+{
+    
+}
+
+std::vector<std::string> GossipNode::connectTCP(std::vector<string> &ips)
```
Comment (zhou-1) [inline src/CPISync/GossipNode.cc:199]: Nice

**Review by zhou-1 — COMMENTED** [review]
Awesome!

**Review by thezhe — APPROVED** [review]
Code looks much cleaner. Adding to it now.

**Review by aktrinh17 — APPROVED** [review]

**Pull request merged**